### ISSUE TITLE
Change filter_post_meta_for_previews `meta_key` to allow for nulls.

### ISSUE THREAD

**Issue by izzygld** [wp-graphql/wp-graphql#1864]
Can we modify the `meta_key` to allow for `null` at  [https://github.com/wp-graphql/wp-graphql/blob/cc0b383259383898c3a1bebe65adf1140290b37e/src/Utils/Preview.php#L22](https://github.com/wp-graphql/wp-graphql/blob/cc0b383259383898c3a1bebe65adf1140290b37e/src/Utils/Preview.php#L22)

Reason:
We Activated the ["Cloudinary – Image and Video Optimization, Manipulation, and Delivery" Plugin](https://en-gb.wordpress.org/plugins/cloudinary-image-management-and-manipulation-in-the-cloud-cdn/)

Error we received:

> 127.0.0.1 -  22/Apr/2021:07:58:50 +0000 "POST /wp/wp-admin/admin-ajax.php" 500
> 2021/04/22 07:58:51 [error] 10#10: *173 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught TypeError: Argument 3 passed to WPGraphQL\Utils\Preview::filter_post_meta_for_previews() must be of the type string, null given, called in /share/public/wp/wp-includes/class-wp-hook.php on line 294 and defined in /share/public/content/plugins/wp-graphql/src/Utils/Preview.php:22
> Stack trace:
> #0 /share/public/wp/wp-includes/class-wp-hook.php(294): WPGraphQL\Utils\Preview::filter_post_meta_for_previews(NULL, 21400, NULL, false)
> #1 /share/public/wp/wp-includes/plugin.php(212): WP_Hook->apply_filters(NULL, Array)
> #2 /share/public/wp/wp-includes/meta.php(564): apply_filters('get_post_metada...', NULL, 21400, NULL, false, 'post')
> #3 /share/public/wp/wp-includes/meta.php(506): get_metadata_raw('post', 21400, NULL, false)
> #4 /share/public/wp/wp-includes/post.php(2226): get_metadata('post', 21400, NULL, false)
> #5 /share/public/content/plugins/cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-media.php(1837): get_post_meta(21400, NULL, false)

**Comment by jasonbahl** [comment]
🤔  How is Cloudinary storing meta with `null` as the meta_key? 🤔 

I'll look into this.

**Comment by izzygld** [comment]
Any ideas?

**Comment by jasonbahl** [comment]
I believe this is also related: https://wordpress.org/support/topic/wp-engine-sites-crash-when-activating-plugin/#post-14439430

**Comment by jasonbahl** [comment]
This may or may not be helpful, but I believe to replicate a GraphQL Query needs to query a field that has a call to `get_post_meta()` and the query needs to be for a preview post. 

I'm thinking we can test like so: 

```php
add_action( 'graphql_register_types', function() {

	update_post_meta( 2179, 'some_meta_key', null );

	register_graphql_field( 'Post', 'metaField', [
		'type' => 'String',
		'resolve' => function( $post ) {
			return get_post_meta( $post->databaseId, 'some_meta_key', true );
		}
	]);

} );
```

Then, run a query like so: 

```graphql
{
  post(id: 2179, idType: DATABASE_ID, asPreview: true) {
    id
    databaseId
    title
    isPreview
    metaField
  }
}
```

I believe, this query, for a preview of a post, and a field that calls `get_post_meta()` _should_ trigger the filter, and ideally should get us the same error being reported here, if I've done things correctly. 

In my quick test, I wasn't able to reproduce, but I think it's headed in the right-ish direction? 

----

@izzygld do you have info on what specific query(s) were leading to the error? Or was it happening with the plugin just simply being installed?

**Comment by jasonbahl** [comment]
oh gosh. . .re-reading the OP, it's about the `meta_key` being null, and I was thinking it was the `meta_value`. . .I totally went down the wrong path

🤦🏻‍♂️ 

----

Reproducing should actually be as simple as running: 

```
add_action( 'init', function() {
get_post_meta( 123, null, true );
});
```

🤔

**Comment by jasonbahl** [comment]
@markkelnar yep, I've reproduced by adding this snippet: 

```php
add_action( 'init', function() {
  get_post_meta( 123, null, true );
});
```

![Screen Shot 2021-05-14 at 12 04 23 PM](https://user-images.githubusercontent.com/1260765/118311031-9ad5a780-b4ac-11eb-86b0-6f347abe4b43.png)

**Comment by izzygld** [comment]
Just a status update:

Heard back from Cloudinary Developers:

> This will be fixed in 2.8.0 that is currently in QA. We hope to release it this week or early next week.
